src/Elastic.Ingest/ShipTo.cs

```diff
@@ -85,9 +82,8 @@ public ShipTo(string cloudId, string username, string password)
 					var basicAuthCredentials = new BasicAuthenticationCredentials(Username, Password);
 					return new CloudConnectionPool(CloudId, basicAuthCredentials);
 				default:
-					return null;
+					throw new NotImplementedException($"Unrecognised connection pool type '{ConnectionPool}' specified in the configuration.");
```

Comment:
Maybe `ArgumentException` is a good fit for this?

Reply:
Changed.